packages/widget/src/connect.ts

```diff
@@ -0,0 +1,129 @@
+import { consume } from '@lit/context';
+import { LitElement, html } from 'lit';
+import { customElement, state } from 'lit/decorators.js';
+import {
+  WalletManagerContext,
+  WalletManagerController
+} from '@buildwithsygma/sygmaprotocol-wallet-manager';
+
+import {
+  SdkManagerContext,
+  SdkManager
+} from '@buildwithsygma/sygmaprotocol-sdk-manager';
+
+import { Environment } from '@buildwithsygma/sygma-sdk-core';
+
+@customElement('connect-dialog')
+class ConnectDialog extends LitElement {
+  @consume({ context: WalletManagerContext, subscribe: true })
+  @state()
+  walletManager?: WalletManagerController;
+
+  @consume({ context: SdkManagerContext, subscribe: true })
+  @state()
+  sdkManager?: SdkManager;
+
+  @state()
+  chainId?: number;
+
+  async connectedCallback(): Promise<void> {
+    super.connectedCallback();
+    this.chainId = (
+      await this.walletManager?.evmWallet?.web3Provider?.getNetwork()
+    )?.chainId;
```

Comment:
all these optional props cause a looot of optional chaining which is a bit misleading (if they are redundant) or fragile (if everything is truly optional)